test/e2e/federatedhpa_test.go

```diff
@@ -0,0 +1,130 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/utils/pointer"
+
+	autoscalingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/autoscaling/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	testhelper "github.com/karmada-io/karmada/test/helper"
+)
+
+var _ = ginkgo.Describe("testing for FederatedHPA and metrics-adapter", func() {
+	var namespace string
+	var deploymentName, serviceName, policyName, federatedHPAName, pressureToolPodName, targetCluster string
+	var deployment *appsv1.Deployment
+	var service *corev1.Service
+	var policy *policyv1alpha1.ClusterPropagationPolicy
+	var federatedHPA *autoscalingv1alpha1.FederatedHPA
+	var pressureTool *corev1.Pod
+
+	ginkgo.BeforeEach(func() {
+		namespace = testNamespace
+		deploymentName = deploymentNamePrefix + rand.String(RandomStrLength)
+		serviceName = deploymentName
+		policyName = deploymentName
+		federatedHPAName = deploymentName
+		pressureToolPodName = deploymentName
+		targetCluster = framework.ClusterNames()[0]
+
+		deployment = testhelper.NewDeployment(namespace, deploymentName)
+		deployment.Spec.Replicas = pointer.Int32(1)
+		deployment.Spec.Template.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+			Limits: map[corev1.ResourceName]resource.Quantity{
+				corev1.ResourceCPU:    resource.MustParse("20m"),
+				corev1.ResourceMemory: resource.MustParse("20Mi"),
+			}}
+		service = testhelper.NewService(namespace, serviceName, corev1.ServiceTypeNodePort)
+		pressureTool = testhelper.NewPod(namespace, pressureToolPodName)
+		pressureTool.Spec.Containers = []corev1.Container{
+			{
+				Name:    pressureToolPodName,
+				Image:   "alpine:3.19.1",
+				Command: []string{"/bin/sh"},
+				Args:    []string{"-c", "apk add curl; while true; do for i in `seq 200`; do curl http://" + serviceName + "." + namespace + ":80; done; sleep 1; done"},
+			},
+		}
+		policy = testhelper.NewClusterPropagationPolicy(policyName, []policyv1alpha1.ResourceSelector{
+			{
+				APIVersion: deployment.APIVersion,
+				Kind:       deployment.Kind,
+				Name:       deploymentName,
+			},
+			{
+				APIVersion: service.APIVersion,
+				Kind:       service.Kind,
+				Name:       serviceName,
+			}, {
+				APIVersion: pressureTool.APIVersion,
+				Kind:       pressureTool.Kind,
+				Name:       pressureToolPodName,
+			}}, policyv1alpha1.Placement{
+			ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+				ClusterNames: []string{targetCluster},
```

Comment:
Generally LGTM, but it would be better if we could specify the `scheduling strategy` here, and be able to set `ClusterNames` to `framework.ClusterNames()`.

Reply:
Do we need to write a test case for each of the different scheduling strategies?

Reply:
> Do we need to write a test case for each of the different scheduling strategies?

That's even better!

Reply:
I've re-pushed a commit to support this, it's a little more complex than the previous one, could help to check it again~